lib/mappers/bhl_mods.py

```diff
@@ -0,0 +1,210 @@
+                title_prop = "titleInfo/title"
+                if exists(relatedItem, title_prop):
+                    _dict["relation"].append(getprop(relatedItem, title_prop))
+
+            self.update_source_resource(self.clean_dict(_dict))
+
+
+    def map_rights(self):
+        prop = self.root_key + "accessCondition/#text"
+
+        if exists(self.provider_data, prop):
+            self.update_source_resource(
+                {"rights": getprop(self.provider_data, prop)}
+            )
+
+    def map_type(self):
+        prop = self.root_key + "typeOfResource"
+
+        if exists(self.provider_data, prop):
+            self.update_source_resource({"type":
+                                             getprop(self.provider_data, prop)})
+
+    def map_edm_has_type(self):
+        prop = self.root_key + "genre"
+        if exists(self.provider_data, prop):
+            genre = getprop(self.provider_data, prop)
+        if "authority" in genre and genre["authority"] == "marcgt":
+            self.update_source_resource({"hasType": genre["#text"]})
+
+    def map_data_provider(self):
+        prop = self.root_key + "note"
+        _dict = {"dataprovider": None}
+
+        if exists(self.provider_data, prop):
+            for s in iterify(getprop(self.provider_data, prop)):
+                try:
+                    if s.get("type") == "ownership":
+                        _dict["dataProvider"] = s.get("#text")
+                        break
+                except:
+                    logger.error("Error getting note/type for record %s" %
+                                 self.provider_data["_id"])
+
+            self.mapped_data.update(self.clean_dict(_dict))
+
+    def map_preview_and_is_shown_at(self):
+        prop = self.root_key + "location/url"
+        ret_dict = {}
+
+        if exists(self.provider_data, prop):
+            for url in iterify(getprop(self.provider_data, prop)):
+                usage = getprop(url, "usage", True)
+                access = getprop(url, "access", True)
+                if (usage == "primary"
+                        and access == "raw object"):
+                    ret_dict["isShownAt"] = url["#text"]
+                elif (access == "object in context"
+                        and usage == "primary display"):
+                    ret_dict["preview"] = url["#text"]
+
+        self.mapped_data.update(ret_dict)
+
+    def map_title(self):
+        path = self.root_key + "titleInfo"
+        titles = []
+        if exists(self.provider_data, path):
+            for titleInfo in getprop(self.provider_data, path):
+                if isinstance(titleInfo, str):
+                    titles.append(titleInfo)
+                if isinstance(titleInfo, dict) \
+                        and "title" in titleInfo \
+                        and not "type" in titleInfo:
+                    try:
+                        titles.append(titleInfo["title"])
+                    except:
```

Comment:
This is probably an artifact of my having taken this code over and was only fixing some stuff up rather than writing it from scratch.